src/sql/workbench/browser/parts/editor/editorStatusModeSelect.ts

```diff
@@ -41,7 +42,7 @@ export async function setMode(accessor: ServicesAccessor, modeSupport: IModeSupp
 		if (newInputCreator) { // if we know how to handle the new language, tranform the input and replace the editor (e.x notebook, sql, etc)
 			const newInput = newInputCreator.convertInput(input || activeEditor);
 			if (newInput) {  // the factory will return undefined if it doesn't know how to handle the input
-				await editorService.replaceEditors([{ editor: activeEditor, replacement: newInput }], activeControl.group);
+				await editorService.replaceEditors([{ editor: activeEditor, replacement: isThenable(newInput) ? await newInput : newInput }], activeControl.group);
```

Comment:
We could just add await here, right? It just wouldn't do anything for the non-promise input. (I don't really have a preference - just confirming that my understanding is correct here)

Reply:
you can't await something that isn't a promise.

Reply:
Is the strict compile blocking that or something? It's perfectly fine to await a non-promise value in normal Typescript. It just doesn't do anything useful. But it's not like we're really gaining anything by using isThenable here.

https://github.com/Microsoft/TypeScript/issues/8310

Reply:
maybe i was interpreting the error i was getting wrong, ill take a look